### DISCUSSION TITLE
Bottleneck of seaweedfs

### DISCUSSION THREAD

**Discussion by PedroMPagani** [seaweedfs/seaweedfs#4379] [Q&A]
What is the bottleneck of seaweed if I run master, filer & volume, I am going to be putting out some average sized files of 2-4MB each, whats the average latency to get the data from one of these files? also, the Java client currently supports only FilerClient iirc, so, would bandwidth be a big problem on the FilerClient? I will probably have around 250k of these files, is there a recommended setup for something like this scenario? such as minimum 3 masters 2 filers etc..?

**Comment by washcycle** [comment]
From my understanding the filer is an index, the client gets the URL/Path to download from the volume server and returns it to the client. In other words you download it from the volume server not the filer. So more volume servers you have the faster your seaweedfs instance will be of course you have to consider disk IO throughput. I think the default filer can handle a huge number of requests, but I keep 2 up for HA.

**Comment by PedroMPagani** [comment]
I ran a small benchmark with files of 7MB, and these were the results: (1000 files write & read)

Size: 7043530

Why would writing take too long? the files are only 7MB, disk is NVME, maybe its network bottleneck? any debugs or commands infos etc i could use to monitor why it takes a long time?

52642.99833800005ms write (52ms)
12979.843437999993ms read (12ms)

@washcycle

**Reply by washcycle** [reply]
How are you launching the services? 'weed server' or separately?

Configuring the volume server can be tricky so those settings might shed light on the situation well.

**Reply by PedroMPagani** [reply]
@washcycle seaweed was setup at once with the weed server, not each component separately 


Such as:
just going off the command of `weed server -dir=/mnt/disk/seaweed -volune.port=8081`

Nothing was really tweaked on any type of config


I've ran some more benchmarks:

Size: 1000
1143
160527.551233
1549.663039

Reading doesnt take much, but writing is really taking a lot...

**Comment by PedroMPagani** [comment]
@washcycle I've discovered some bottlenecks on the Java client, which I fixed myself but now, the best way to know the RAW cost of these operations should be through the weed benchmark?

**Comment by PedroMPagani** [comment]
Did some more investigation and endedup finding the main cost of this, disk speed, the disk is an nvme and i thought for some reason that it should be faster, I guess i was wrong
